### PR TITLE
ENT-4973: Altering modal's scroll, adding course availability dates

### PR DESCRIPTION
**Pull request by kiram15**
Screenshots also available[ in the ticket ](https://openedx.atlassian.net/secure/RapidBoard.jspa?rapidView=630&projectKey=ENT&modal=detail&selectedIssue=ENT-4973)for clarity, cleared by UX (Jenn)

### CODE REVIEW

File: src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx [hunk 1]
```diff
@@ -38,55 +47,56 @@ const CatalogCourseInfoModal = ({
...
+                  {intl.formatMessage(messages['catalogCourseInfoModal.educationBadge'])}
+                </Badge>
+              )}
+            </div>
+            <p className="course-info-description-title">
+              {intl.formatMessage(messages['catalogCourseInfoModal.courseDescriptionTitle'])}
+            </p>
+            <p className="course-info-description-subtitle">
+              {intl.formatMessage(messages['catalogCourseInfoModal.availabilityMessage'])}
+              {formatDate({ startDate }, { endDate })}
```
Comment (binodpant) [inline src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx]: this is a curious input form? should it be `formatDate(startDate, endDate)` instead? alternatively you can also go with `formatDate({startDate, endDate}) as long as you destructure in the function def.

File: src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx [hunk 2]
```diff
@@ -14,6 +14,12 @@ import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
 import { Launch } from '@edx/paragon/icons';
 import messages from './CatalogCourseInfoModal.messages';
 
+function formatDate(start, end) {
+  const startDate = (new Date(start.startDate)).toLocaleDateString();
```
Comment (binodpant) [inline src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx]: based on the proptypes, looks like `start` and `end` should be strings and not objects?

File: src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx [hunk 3]
```diff
@@ -14,6 +14,12 @@ import { injectIntl, intlShape } from '@edx/frontend-platform/i18n';
...
+  const endDate = (new Date(end.endDate)).toLocaleDateString();
+  return ((` ${startDate} - ${endDate}`));
```
Comment (binodpant) [inline src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx]: is this extra leading space actually needed or?

Comment (binodpant) [reply src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx]: I think the outer parens are superflous, maybe just try :

```
return `${startDate} - ${endDate}`
```

File: src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx (hunk 2 above)
Comment (alex-sheehan-edx) [reply src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx:18]: I was testing with her- printing out the variables interestingly yielded dicts, maybe the proptype should be changed?

File: src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx [hunk 4]
```diff
@@ -38,55 +47,56 @@ const CatalogCourseInfoModal = ({
...
         className="course-info-modal"
       >
-        <ModalDialog.Hero className="course-info-hero">
-          <ModalDialog.Hero.Background backgroundSrc={bannerImageUrl} />
-        </ModalDialog.Hero>
-
-        <ModalDialog.Header className="course-info-modal-header">
-
+        <ModalDialog.Body className="full-body">
+          <ModalDialog.Hero>
```
Comment (alex-sheehan-edx) [inline src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx:51]: love that you can just throw this into the body

File: src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx (hunk 1 above)
Comment (alex-sheehan-edx) [reply src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx]: 😄 I had spotted the same thing before seeing your comment- this is why https://github.com/edx/frontend-app-enterprise-public-catalog/pull/124/files#diff-caee6ed0461fad8f827d7e53ce7b4f859c9ff39f02d45323ed7d6d8bda2d1cd6R18 is an object

**Review by alex-sheehan-edx — APPROVED** [review]
I agree with @binodpant's comments, + I think you have params to tweak but otherwise this looks great 😄

File: src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx (hunk 3 above)
Comment (kiram15) [reply src/components/catalogCourseInfoModal/CatalogCourseInfoModal.jsx]: I have the extra leading space just to give it a space after the dash in availability, but I can put that in the message if that makes more sense!

**Pull request merged**